assets/src/components/detours/detourMap.tsx

```diff
@@ -103,7 +110,7 @@ export const DetourMap = ({
   waypoints,
 
   onClickOriginalShape,
-  onClickMap,
+  onAddWaypoint,
```

Comment:
Soooooo I was a bit torn about this, but here's some of my thinking:

- Originally, we were treating this as `onClickMap`, because we didn't want `<DetourMap />` to be overly concerned with what it means to actually click the map. Basically, we were letting the mapping of "click the map" <-> "add waypoint" be `<DiversionPage />`'s responsibility.
- BUT
- There's more complicated logic now, which means that to keep `<DetourMap />` agnostic to what clicking the map means would force that logic into `<DiversionPage />` (including pushing street view stuff up into `<DiversionPage />`, which ...I don't want to do.
- And it doesn't bother me that much to have `<DetourMap />` be responsible for that. I get that having the `<Map />` be in charge of both "how do I recognize a click?" and "what should I do when you click?" is an SRP violation, and it's better to have some additional layer to separate those questions, but `<DetourMap />` IS that layer.
- Basically, I don't think it's a problem to let `<DetourMap />` know things about detours.
- That said, it would be good to give `onClickOriginalShape` the same treatment as `onClick` -> `onAddWaypoint`. Right now, `onAddWaypoint` and `onClickOriginalShape` are sort of living at two separate layers of abstraction, which is not the best thing in the world.
- I didn't do that here for two reasons. First, I didn't need to. And second, `onClick` was a bit more nicely encapsulated, which made this change easy. `onClickOriginalShape` (and its friends `startPoint === null` / `endPoint === null`) is (are) more tangled, and I'd rather muck around with them as a follow up.